### ISSUE TITLE
Release trained models from Uniform Loss vs. Specialized Optimization on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [Gabriel-SGama/UnitScal_vs_SMTOs#1]
Hi @Gabriel-SGama 🤗

Niels here from the open-source team at Hugging Face. I discovered your work on Arxiv and was wondering whether you would like to submit it to [hf.co/papers](https://hf.co/papers) to improve its discoverability. If you are one of the authors, you can submit it at https://huggingface.co/papers/submit.

The paper page lets people discuss about your paper and lets them find artifacts about it (your models for instance), you can also claim the paper as yours which will show up on your public profile at HF, add Github and project page URLs.

It'd be great to make the trained model checkpoints from your comparative analysis available on the 🤗 hub, to improve their discoverability/visibility. I noticed you currently host them on Google Drive. We can add tags so that people find them when filtering https://huggingface.co/models.

## Uploading models

See here for a guide: https://huggingface.co/docs/hub/models-uploading.

In this case, we could leverage the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which adds `from_pretrained` and `push_to_hub` to any custom `nn.Module`. Alternatively, one can leverages the [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file) one-liner to download a checkpoint from the hub.

We encourage researchers to push each model checkpoint (e.g., for different tasks/datasets like MNIST, Cityscapes, and QM9, and different optimizers if applicable) to a separate model repository, so that things like download stats also work. We can then also link the checkpoints to the paper page.

You can also build a demo for your models on [Spaces](https://huggingface.co/spaces), we can provide you a ZeroGPU [grant](https://huggingface.co/docs/hub/en/spaces-gpus#community-gpu-grants), which gives you A100 GPUs for free.

Let me know if you're interested/need any help regarding this!

Cheers,

Niels
ML Engineer @ HF 🤗

**Comment by Gabriel-SGama** [comment]
Hi @NielsRogge,

Thank you for showing interest in our paper, I would be interested in submitting the paper to Hugging Face. However, the data folder that I made available only contains the training/evaluation metrics of each run. I didn't save the model checkpoints as it would take up too much disk space due to the grid search, and even if I had saved them I wouldn't be able to make them publicly available. It is possible to replicate the results and get those checkpoints (they might differ slightly from the results published on the paper).

Can you give me directions related to the model checkpoints? As our paper is about a comparative analysis of multiple Specialized Multi-Task Optimizers (SMTOs) and not a particular application, would it be helpful to publish all checkpoints from the optimal hyperparameters of each SMTO in each dataset? Or would it be better to just show an example that demonstrates the core idea of Multi-Task Learning? I appreciate any advice you might have for me!

Best,

Gabriel Gama
